### PR TITLE
Address CRAN comments

### LINKED ISSUE #946: Address CRAN comments
> Thanks,
If there are references describing the methods in your package, please
add these in the description field of your DESCRIPTION file in the form
authors (year) <doi:...>
authors (year) <arXiv:...>
authors (year, ISBN:...)
or if those are not available: <https:...>
with no space after 'doi:', 'arXiv:', 'https:' and angle brackets for
auto-linking.
(If you want to add a title as well please put it in quotes: "Title")
Please add \value to .Rd files regarding exported methods and explain
the functions results in the documentation. Please write about the
structure of the output (class) and also what the output means. (If a
function does not return a value, please document that too, e.g.
\value{No return value, called for side effects} or similar)
Missing Rd-tags in up to 35 .Rd files, e.g.:
      assert_list_element.Rd: \value
      assert_one_to_one.Rd: \value
      assert_varval_list.Rd: \value
      ...
You have examples for unexported functions.
Please either omit these examples or export these functions.
Used ::: in documentation:
      man/arg_name.Rd:
         test_fun <- function(something) {
  admiral:::arg_name(substitute(something))
}
etc..
Please fix and resubmit.

### PR DESCRIPTION
**Pull request by thomas-neitmann**
Close #946 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

File: R/assertthat_checks.R [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+is_vars <- function(arg) {
+  inherits(arg, "quosures") && all(map_lgl(arg, quo_is_symbol))
+}
+on_failure(is_vars) <- function(call, env) {
+  paste0(
+    "Argument `",
+    deparse(call$arg),
+    "` is not a list of variables created using `vars()`"
+  )
+}
```
Comment (bundfussr) [inline R/assertthat_checks.R:271]: This function is called in `extract_duplicate_records()` and `signal_duplicate_records()` only.

Should we replace the calls by `assert_vars()` and remove `is_vars()` from admiral?

**Comment by thomas-neitmann** [comment]
@pharmaverse/admiral-devs I think I addressed all CRAN comments. Please check. Thank you!

File: R/derive_param_tte.R [hunk 2]
```diff
@@ -669,32 +669,6 @@ extend_source_datasets <- function(source_datasets,
 #' @seealso [derive_param_tte()], [censor_source()], [event_source()]
 #'
 #' @return An object of class `tte_source`
```
Comment (bms63) [inline R/derive_param_tte.R:671]: How come we are taking out the example?

Comment (thomas-neitmann) [reply R/derive_param_tte.R:671]: They didn't like un-exported functions having examples where you make use of `:::`.

File: R/assertthat_checks.R [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+#' @return `TRUE` if the argument is a date or date-time, `FALSE` otherwise
+#'
+#' @keywords check
+#'
+#' @noRd
+#'
+#' @examples
+#' refdate <- lubridate::ymd("2020-01-02")
+#' date <- lubridate::ymd("2020-02-03")
+#' assertthat::assert_that(admiral:::is_date(refdate), admiral:::is_date(date))
```
Comment (bms63) [inline R/assertthat_checks.R:18]: Is this `:::` since this is exported?

Comment (thomas-neitmann) [reply R/assertthat_checks.R:18]: For all these `is_*()` function I added the `@noRd` tag which means no actual documentation pages will be created. I want to get rid of these functions any, we should no longer make use of `assertthat::assert_that()`.

**Review by bms63 — APPROVED** [review]
Okay I think I understand these changes

File: R/assertthat_checks.R [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+  # Note: is.language allows both symbol and language
+  !is.list(arg) & is.language(arg)
+}
+on_failure(is_expr) <- function(call, env) {
+  paste0(
+    "Argument `",
+    deparse(call$arg),
+    "` is not an expression created using `expr()`"
+  )
+}
```
[the remark this answers is not in the document]
Comment (thomas-neitmann) [reply R/assertthat_checks.R:320]: If that's indeed the case, then yes.

[the remark this answers is not in the document]
Comment (thomas-neitmann) [reply R/assertthat_checks.R]: Removed.

File: R/assertthat_checks.R (hunk 1 above)
Comment (thomas-neitmann) [reply R/assertthat_checks.R]: Removed.

File: R/test_helpers.R [hunk 5]
```diff
@@ -8,6 +8,9 @@
 #'        `base` and `compare` datasets
 #' @param ... Additional arguments passed onto [diffdf::diffdf()]
 #'
+#' @return
+#' An error if `base` and `compare` do not match of `NULL` invisiblly if they do
```
Comment (bundfussr) [inline R/test_helpers.R:12]: Typos:
- "of" instead of "or"
- "invisiblly" instead of "invisibly"

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: R/test_helpers.R (hunk 5 above)
Comment (thomas-neitmann) [reply R/test_helpers.R]: Fixed.

**Review by bundfussr — APPROVED** [review]

**Comment by bundfussr** [comment]
To fix the failing checks I think we should implement the `accept_grouped` parameter for `assert_data_frame()` as discussed in https://github.com/pharmaverse/admiral/pull/610.

**Comment by thomas-neitmann** [comment]
I called it `check_is_grouped` but it's essentially the same. Just checking locally and then I'm going to push.

**Pull request merged**